rust/lance/src/dataset/optimize.rs

```diff
@@ -755,11 +755,11 @@ async fn rewrite_files(
         reserve_fragment_ids(&dataset, new_fragments.iter_mut()).await?;
 
         if options.defer_index_remap {
-            let mut changed_row_addrs = Vec::with_capacity(row_ids.serialized_size());
-            row_ids.serialize_into(&mut changed_row_addrs)?;
+            let mut changed_row_addrs = Vec::with_capacity(row_addrs.serialized_size());
+            row_addrs.serialize_into(&mut changed_row_addrs)?;
             (None, Some(changed_row_addrs))
         } else {
-            let row_id_map = remapping::transpose_row_ids(row_ids, &fragments, &new_fragments);
+            let row_id_map = remapping::transpose_row_ids(row_addrs, &fragments, &new_fragments);
```

Comment:
@jackye1995 Shall we rename this to `transpose_row_addrs`?

Reply:
yeah I think that makes sense, the stable row ID path does not use this anyway it's purposely used for row address